### ISSUE TITLE
Issue with link '/' stripping?

### ISSUE THREAD

**Issue by choldgraf** [jupyterhub/jupyter-server-proxy#37]
In https://github.com/binder-examples/r/issues/13 we found some strange behavior. It seems like you get drastically different link resolving if there is a `/` at the end of the URL or not when using nbserverproxy. For example, the following link:

```
http://mybinder.org/v2/gh/binder-examples/r/master?urlpath=shiny/bus-dashboard
```

resolves incorrectly to 

https://hub.mybinder.org/hub/bus-dashboard/?token=<MYTOKEN>

while the same link with a slash at the end:

```
http://mybinder.org/v2/gh/binder-examples/r/master?urlpath=shiny/bus-dashboard/
```

resolves correctly to 

```
https://hub.mybinder.org/user/binder-examples-r-2i4xen56/shiny/bus-dashboard/?token=<MYTOKEN>
```

Is this expected behavior?

**Comment by ryanlovett** [comment]
No, that's not the expected behavior. :) I'm a bit occupied with Summer courses atm, but can take a look at this in a week or so.

**Comment by betatim** [comment]
It might be as easy as figuring out the correct regex to put in the [handler registration](https://github.com/jupyterhub/nbserverproxy/blob/c2b11887232d41d8cfd2469585cfbc09001a4a8a/nbserverproxy/handlers.py#L469). At least that is where I would start.
